### PR TITLE
Spec: Permissions policy integration

### PR DESCRIPTION
**Pull request by alexmturner**
Defines the "private-aggregation" permissions policy and plumbs values through. Improves other algorithm's plumbing as a drive-by.

***
<a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/pull/66.html" title="Last updated on Jun 20, 2023, 7:52 PM UTC (e1e9da0)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/patcg-individual-drafts/private-aggregation-api/66/270f49e...e1e9da0.html" title="Last updated on Jun 20, 2023, 7:52 PM UTC (e1e9da0)">Diff</a>

### CODE REVIEW

**Comment by alexmturner** [comment]
@linnan-github, could you PTAL? Thanks!
cc @yoavweiss

File: spec.bs [hunk 1]
```diff
@@ -916,9 +929,13 @@ as appropriate:
     [=upon rejection=] of |promise|, run the following steps:
     1. Let |globalScopes| be |this|'s [=Worklet/global scopes=].
     1. [=Assert=]: |globalScopes|' [=list/size=] equals 1.
-    1. Set |globalScopes|[0]'s {{SharedStorageWorkletGlobalScope/
-        privateAggregation}}'s [=PrivateAggregation/scoping details=] a new
-            [=scoping details=] with the items:
+    1. Let |privateAggregationObj| be |globalScopes|[0]'s
+        {{SharedStorageWorkletGlobalScope/privateAggregation}}.
+    1. If [=this=]'s [=relevant global object=]'s [=associated document=] is
```
Comment (linnan-github) [inline spec.bs:934]: Optional: We can probably just set this field to the permission policy check without checking the value.

File: spec.bs [hunk 2]
```diff
@@ -916,9 +929,13 @@ as appropriate:
...
+        [=allowed to use=] "<code>[=private-aggregation=]</code>", set
+        |privateAggregationObj|'s [=PrivateAggregation/allowed to use=] to true.
+    1. Set |privateAggregationObj|'s [=PrivateAggregation/scoping details=] a
```
Comment (linnan-github) [inline spec.bs:937]:
```suggestion
    1. Set |privateAggregationObj|'s [=PrivateAggregation/scoping details=] to a
```

Comment (linnan-github) [inline spec.bs:937]: Shall we early return if it's not allowed to use?

File: spec.bs [hunk 3]
```diff
@@ -1071,35 +1090,49 @@ nested scope), renumbering this and later steps as necessary.
...
+|auctionConfig| and an [=interest group=] or null |ig|.
+
+Second, we add the following steps after step 11 ("If
 <var ignore>evaluationStatus</var> is an [=ECMAScript/abrupt completion=]..."),
 renumbering later steps as appropriate:
-<div algorithm="protected-audience-evaluate-a-script-monkey-patch-1">
-12. Let |debugScope| be a new [=debug scope=].
-1. Set <var ignore>global</var>'s {{InterestGroupScriptRunnerGlobalScope/
+<div algorithm="protected-audience-evaluate-a-script-monkey-patch">
+12. If |auctionConfig|'s [=auction config/permissions policy state=]'s
```
Comment (linnan-github) [inline spec.bs:1101]: Similar to shared storage, probably can just set the field without checking the value, and early return?

File: spec.bs (hunk 2 above)
Comment (alexmturner) [reply spec.bs:937]: The current (implemented) behavior is actually to check the 'is initial execution' complete first, although we could consider swapping.

File: spec.bs (hunk 1 above)
Comment (alexmturner) [reply spec.bs]: Done

File: spec.bs (hunk 2 above)
Comment (alexmturner) [reply spec.bs]: Done

File: spec.bs (hunk 3 above)
Comment (alexmturner) [reply spec.bs]: Switched the setting, but left early return as above

File: spec.bs (hunk 2 above)
Comment (alexmturner) [reply spec.bs]: Yeah, that's fair -- I've added an issue to track whether we want to swap the order. I'll follow up soon

**Review by linnan-github — APPROVED** [review]

**Comment by alexmturner** [comment]
Thanks!

**Pull request merged**